.tekton/scripts/check-partner-tasks.sh

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$SCRIPTDIR/../.." || exit 1
+
+if [ ! -e "partners/" ]; then
+    echo "No partners directory exists. Skip checks against partner tasks."
+    exit 0
+fi
+
+check_dir_structure() {
+    resultf=$(mktemp)
+    check_result=$(mktemp)
+
+    for task_dir in partners/*; do
+        owners_file="$task_dir/OWNERS"
+        if [ ! -e "$owners_file" ]; then
+            echo "error: missing file $owners_file" >>"$check_result"
+            echo "issue found" >"$resultf"
+        fi
+
+        task_name=$(basename "$task_dir")
+        find "$task_dir" -mindepth 1 -maxdepth 1 -type d | \
+        while read -r version_dir; do
+            task_version=$(basename "$version_dir")
+            if ! echo "$task_version" | grep "^[0-9]\+\.[0-9]\+$" >/dev/null 2>&1; then
```

Comment:
```
error: partners/test/1.42.2 is not a version directory. A version directory name only contains digits and dot, e.g. 0.1, 1.2
```
Any reason why limit only to major.minor?

Reply:
No special reason, just referenced the version used in tekton catalog back to the time when catalog was introduced to build-definitions. In practice, when task API is changed, it works well by increasing either major or minor version, that should depend on the preference of task maintainer. `major.minor` looks enough for the task version maintenance purpose.

Reply:
Oh, if there is a reason, that is to keep consistency with the version scheme used for predefined tasks under `task/`, at where version scheme is convention by task maintainers and not forced.